includes/admin/class-admin.php

```diff
@@ -0,0 +1,218 @@
+	 * 2) sets up plugin settings and displays admin page content
+	 */
+	protected function __construct() {
+
+		// Setup properties used throughout this class
+		$this->page_slug   = 'hm-toc-settings';
+		$this->option_slug = 'hm_content_toc';
+		$this->page_title  = __( 'HM Content TOC Settings', 'hm-content-toc' );
+
+		// Add admin submenu page to Settings
+		add_action( 'admin_menu', array( $this, 'add_plugin_option_menu_page' ) );
+
+		// Setup plugin settings and display admin page content
+		add_action( 'admin_init', array( $this, 'setup_plugin_option_settings' ) );
+	}
+
+	/**
+	 * Make class a singleton, as we don't need more than
+	 * one instance of it
+	 *
+	 * @return Admin True single instance of the class
+	 */
+	public static function get_instance() {
+
+		static $instance = null;
+
+		if ( is_null( $instance ) ) {
+			$instance = new static;
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Adds option submenu page to WP Settings page
+	 * Only users with capability `manage_options` will see it
+	 */
+	public function add_plugin_option_menu_page() {
+
+		add_options_page(
+			$this->page_title,
+			$this->page_title,
+			'manage_options',
+			$this->page_slug,
+			array( $this, 'display_settings_page' )
+		);
+	}
+
+	/**
+	 * Setup plugin admin:
+	 * 1) Register plugin option
+	 * 2) Register settings section and fields
+	 */
+	public function setup_plugin_option_settings() {
+
+		// Register settings for plugin option
+		register_setting(
+			$this->option_slug,
+			$this->option_slug,
+			array( $this, 'option_sanitise' )
+		);
+
+		// Add settings section - anonymous, so just we can add fields to it
+		add_settings_section(
+			'hm-toc-section',
+			'',
+			'__return_empty_string',
+			$this->page_slug
+		);
+
+		// Plugin input setting fields
+		$input_fields = array(
+			'title'   => array(
+				'name' => __( 'Title', 'hm-content-toc' ),
+				'desc' => __( 'The title is added before generated TOC links', 'hm-content-toc' ),
+			),
+			'headers' => array(
+				'name' => __( 'Header Elements', 'hm-content-toc' ),
+				'desc' => sprintf(
+					__( 'Comma separated list of HTML element names to generate TOC for. For example: %s', 'hm-content-toc' ),
+					'<code>h2, h3, h4</code>'
+				),
+			),
+		);
+
+		// Display input fields
+		foreach ( $input_fields as $field => $labels ) {
+
+			add_settings_field(
+				"hm-toc-{$field}",
+				$labels['name'],
+				array( $this, 'display_input_field' ),
+				$this->page_slug,
+				'hm-toc-section',
+				array(
+					'label_for' => "hm-toc-{$field}",
+					'field'     => $field,
+					'desc'      => $labels['desc'],
+				)
+			);
+		}
+	}
+
+	/**
+	 * Display the content of the plugin admin page
+	 */
+	public function display_settings_page() {
+	?>
+		<div class="wrap">
+
+			<?php
+			// Display settings page title
+			printf(
+				'<h2>%s</h2><p>%s</p>',
+				esc_html( $this->page_title ),
+				wp_kses_post( __( 'Specify default settings for HM Content TOC plugin', 'hm-content-toc' ) )
```

Comment:
This doesn't need a call to `wp_kses_post()` because there's no markup in the string. `esc_html__()` is enough.